gulp/tasks/upload-preview.js

```diff
@@ -0,0 +1,17 @@
+(function() {
+    'use strict';
+
+    var gulp = require('gulp'),
+        gitUtils = require('../util/gitUtils'),
+        childProcess = require('child_process'),
+        previewDomain = 'ux-test.edx.org',
+        previewSiteDir = '_preview_site';
+
+    gulp.task('upload-preview', function() {
+        var branch = gitUtils.currentBranch();
+        childProcess.execSync(
+            'aws s3 sync ' + previewSiteDir + ' s3://' + previewDomain + '/' + branch
```

Comment:
I'm having a harder time understanding what this would look like. How would they plug in their own hosting solution? I'd rather wait until we have a use case to know what this might look like. For now, I think these tasks are just for edX to use internally, which I suppose is a little unclean in an open source repo. If I removed the edx.org constant then having a dependency on S3 seems okay for now. I could rename it upload-s3-preview if you think that's clearer.